tensorflow/stream_executor/rocm/rocm_helpers.cu.cc

```diff
@@ -31,5 +31,16 @@ void broadcast_fp32(void* stream, float* dst, int dst_stride, int batches, int s
   hipLaunchKernelGGL(broadcast_fp32_kernel, dim3(x_blocks, (batches+3)/4, src_batches), min(256, (int)size), 0,
                      (hipStream_t)stream, dst, dst_stride, batches, src, size);
 }
+__global__ void __xla_MakeBatchPointers(char* base, int stride, int n, void** ptrs_out) {
+  int idx = threadIdx.x + blockIdx.x * blockDim.x;
+  if (idx >= n) return;
+  ptrs_out[idx] = base + idx * stride;
+}
+
```

Comment:
add a comment block here explaining what this routine does

Reply:
done.